tests/test_utils.py

```diff
@@ -0,0 +1,198 @@
+import os
+from unittest.mock import Mock, patch
+
+import pytest
+import requests
+from colorama import Fore
+from git import Repo
+
+from autogpt.utils import (
+    clean_input,
+    get_bulletin_from_web,
+    get_current_git_branch,
+    get_latest_bulletin,
+    readable_file_size,
+    validate_yaml_file,
+)
+
+
+def test_validate_yaml_file_valid():
+    with open("valid_test_file.yaml", "w") as f:
+        f.write("setting: value")
+    result, message = validate_yaml_file("valid_test_file.yaml")
+    os.remove("valid_test_file.yaml")
+
+    assert result == True
+    assert "Successfully validated" in message
+
+
+def test_validate_yaml_file_not_found():
+    result, message = validate_yaml_file("non_existent_file.yaml")
+
+    assert result == False
+    assert "wasn't found" in message
+
+
+def test_validate_yaml_file_invalid():
+    with open("invalid_test_file.yaml", "w") as f:
+        f.write(
+            "settings:\n  first_setting: value\n  second_setting: value\n    nested_setting: value\n  third_setting: value\nunindented_setting: value"
+        )
+    result, message = validate_yaml_file("invalid_test_file.yaml")
+    os.remove("invalid_test_file.yaml")
+    print(result)
+    print(message)
+    assert result == False
+    assert "There was an issue while trying to read" in message
+
+
+def test_readable_file_size():
+    size_in_bytes = 1024 * 1024 * 3.5  # 3.5 MB
+    readable_size = readable_file_size(size_in_bytes)
+
+    assert readable_size == "3.50 MB"
+
+
+@patch("requests.get")
+def test_get_bulletin_from_web_success(mock_get):
+    mock_get.return_value.status_code = 200
+    mock_get.return_value.text = "Test bulletin"
+    bulletin = get_bulletin_from_web()
+
+    assert bulletin == "Test bulletin"
+
+
+@patch("requests.get")
+def test_get_bulletin_from_web_failure(mock_get):
+    mock_get.return_value.status_code = 404
+    bulletin = get_bulletin_from_web()
+    print(bulletin)
+    assert bulletin == ""
+
+
+def test_get_current_git_branch():
+    branch_name = get_current_git_branch()
```

Comment:
This won’t work in ci as GitHub actions doesn’t check out the branch but the commit 

see docs here https://github.com/actions/checkout#usage

add a decorator to skip in CI to test/utils.py

Reply:
Do this but in tests/utils.py similar to how the existing one there is to skip if the api key doesn’t exist

Reply:
this seems to be resolved

Reply:
Right, move the snippet you used to skip to the test/utils.py so other tests can use it as it’s super useful! There’s an example of how it’s done in that file